scenes/Player/Player.gd

```diff
@@ -68,10 +69,10 @@ func _apply_collision_knockback_damage()-> bool:
 		
 	return false
 
-func set_selection():
+func handle_landing_request():
 	var planets = get_tree().get_nodes_in_group("Planets")
 	var closest_planet = planets[0]
-	
+	var speed_check = velocity.abs()
```

Comment:
Looks great. Only change I'd make is to use velocity.length() rather than .abs and then adding the x and y components. This will be more accurate in terms of the actual physics.